### ISSUE TITLE
Shared-fn has default value which results app/update-shared! to always be called

### ISSUE THREAD

**Issue by TimovanderKamp** [fulcrologic/fulcro#425]
Hi,

I am trying to optimize our app startup time. What I noticed while looking into the Chrome Perfomance tool is that the function app/update-shared! was called a few times during an app mount even though no :shared-fn was configured.

What I found is that there always will be a shared-fn configured as it defaults to `(contantly {})`

```
(>defn update-shared!
  "Force shared props to be recalculated. This updates the shared props on the app, and future renders will see the
   updated values. This is a no-op if no shared-fn is defined on the app. If you're using React 16+ consider using
   Context instead of shared."
  [{::keys [runtime-atom] :as app}]
  [::app => any?]
  (try
    (when-let [shared-fn (ah/app-algorithm app :shared-fn)]
      (let [shared       (-> app ::runtime-atom deref ::static-shared-props)
            state        (current-state app)
            root-class   (-> app ::runtime-atom deref ::root-class)
            query        (comp/get-query root-class state)
            v            (fdn/db->tree query state state)
            shared-props (merge shared (shared-fn v))]
        (swap! runtime-atom assoc ::shared-props shared-props)))
    (catch #?(:cljs :default :clj Throwable) e
      (log/error e "Cannot compute shared"))))
```

```
{:com.fulcrologic.fulcro.algorithm/tx!                    default-tx!
  :com.fulcrologic.fulcro.algorithm/optimized-render!      (or optimized-render! ident-optimized/render!)
  :com.fulcrologic.fulcro.algorithm/shared-fn              (or shared-fn (constantly {}))
  :com.fulcrologic.fulcro.algorithm/render-root!           render-root!
  :com.fulcrologic.fulcro.algorithm/hydrate-root!          hydrate-root!
  :com.fulcrologic.fulcro.algorithm/unmount-root!          unmount-root!
  :com.fulcrologic.fulcro.algorithm/render!                render!
  :com.fulcrologic.fulcro.algorithm/remote-error?          (or remote-error? default-remote-error?)
  :com.fulcrologic.fulcro.algorithm/global-error-action    global-error-action
  :com.fulcrologic.fulcro.algorithm/merge*                 merge/merge*
  :com.fulcrologic.fulcro.algorithm/default-result-action! (or default-result-action! mut/default-result-action!)
  :com.fulcrologic.fulcro.algorithm/global-eql-transform   (or global-eql-transform default-global-eql-transform)
  :com.fulcrologic.fulcro.algorithm/index-root!            indexing/index-root!
  :com.fulcrologic.fulcro.algorithm/index-component!       indexing/index-component!
  :com.fulcrologic.fulcro.algorithm/drop-component!        indexing/drop-component!
  :com.fulcrologic.fulcro.algorithm/props-middleware       props-middleware
  :com.fulcrologic.fulcro.algorithm/render-middleware      render-middleware
  :com.fulcrologic.fulcro.algorithm/schedule-render!       schedule-render!}         
```

I would love to make a pull request which gets rid of the default `(constantly {})` but I have no idea what the impact would be

**Comment by awkay** [comment]
What is the *actual measured* performance impact?

**Comment by awkay** [comment]
So, I *think* it would be a compatible change to remove the default, BUT you would also have to change `update-shared` so that the static shared props are used "raw" if there is no shared-fn...that would, in fact, be faster...

**Comment by awkay** [comment]
Fixed in 01fd1807bc3b686105a5390025ea27f5b56d8433
